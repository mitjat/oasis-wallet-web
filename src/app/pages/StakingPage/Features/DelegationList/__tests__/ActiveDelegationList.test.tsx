import userEvent from '@testing-library/user-event'
import { screen, waitFor } from '@testing-library/react'
import * as React from 'react'
import { render } from '@testing-library/react'

import { Provider } from 'react-redux'
import { ActiveDelegationList } from '../ActiveDelegationList'
import { configureAppStore } from 'store/configureStore'
import { stakingActions } from 'app/state/staking'

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    }
  },
}))

const renderComponent = store =>
  render(
    <Provider store={store}>
      <ActiveDelegationList />
    </Provider>,
  )

describe('<ActiveDelegationList  />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
  })

  it('should match snapshot', () => {
    const component = renderComponent(store)
    store.dispatch(
      stakingActions.updateDelegations([
        {
          amount: '100',
          shares: '100',
          validatorAddress: 'test-validator',
          validator: {
            commission_schedule: {},
            current_rate: {
              epochStart: 0,
              rate: 0.07,
            },
            address: 'test-validator',
            rank: 1,
            status: 'active',
            name: 'test-validator',
          },
        },
      ]),
    )

    expect(component).toMatchSnapshot()
  })

  it('should expand and display the delegation on click', async () => {
    renderComponent(store)
    store.dispatch(
      stakingActions.updateDelegations([
        {
          amount: '100',
          shares: '100',
          validatorAddress: 'oasis1qqv25adrld8jjquzxzg769689lgf9jxvwgjs8tha',
          validator: {
            address: 'oasis1qqv25adrld8jjquzxzg769689lgf9jxvwgjs8tha',
            commission_schedule: {},
            rank: 1,
            status: 'active',
            name: 'test-validator1',
          },
        },
        { amount: '50', shares: '50', validatorAddress: 'oasis1qq2vzcvxn0js5unsch5me2xz4kr43vcasv0d5eq4' },
      ]),
    )

    let row = screen.getByText(/test-validator1/)
    expect(row).toBeVisible()
    userEvent.click(row)

    const details = await screen.findByTestId('validator-item')
    row = screen.getAllByText(/test-validator1/)[0]

    expect(details).toBeVisible()
    userEvent.click(row)
    await waitFor(() => expect(details).not.toBeVisible())
  })
})
