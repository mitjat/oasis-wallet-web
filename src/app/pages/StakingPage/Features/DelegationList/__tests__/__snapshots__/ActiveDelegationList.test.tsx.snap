// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveDelegationList  /> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c12 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c12 g {
  fill: inherit;
  stroke: inherit;
}

.c12 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c12 *[stroke*="#"],
.c12 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c12 *[fill-rule],
.c12 *[FILL-RULE],
.c12 *[fill*="#"],
.c12 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c18 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #00C781;
  stroke: #00C781;
}

.c18 g {
  fill: inherit;
  stroke: inherit;
}

.c18 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c18 *[stroke*="#"],
.c18 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c18 *[fill-rule],
.c18 *[FILL-RULE],
.c18 *[fill*="#"],
.c18 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c1 {
  margin: 0px;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
}

.c4 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  color: false;
  background-color: false;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  background-color: false;
  min-height: 56px;
  border-bottom-width: 1px;
  border-bottom-color: #AAAAAAaa;
  border-bottom-style: solid;
}

.c7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  font-size: 18px;
  font-weight: 500;
  color: false;
  padding-left: 16px;
  padding-right: 16px;
}

.c16 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: 16px;
  padding-right: 16px;
  word-break: break-word;
}

.c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 34px;
  max-width: 34px;
}

.c9 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
}

.c13 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 100px;
  max-width: 100px;
}

.c17 {
  font-weight: 400;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  width: 100%;
  box-sizing: border-box;
  font-size: 16x;
  color: false;
  background-color: false;
  min-height: 48px;
}

.c15:not(:last-of-type) {
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: #AAAAAAaa;
}

.c15:hover {
  color: false;
  background-color: #88888833;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  border-bottom-color: false;
  outline-style: solid;
  outline-width: 1px;
  outline-color: false;
}

.c15:hover {
  cursor: pointer;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  line-height: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10:focus {
  outline: none;
  color: false;
}

.c10:hover {
  color: false;
}

.c10 span.__rdt_custom_sort_icon__ i,
.c10 span.__rdt_custom_sort_icon__ svg {
  opacity: 0;
  color: inherit;
  font-size: 18px !important;
  height: 18px !important;
  width: 18px !important;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-transition-duration: 125ms;
  transition-duration: 125ms;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c10 span.__rdt_custom_sort_icon__.asc i,
.c10 span.__rdt_custom_sort_icon__.asc svg {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  line-height: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c11:focus {
  outline: none;
  color: false;
}

.c11:hover {
  color: false;
}

.c11 span.__rdt_custom_sort_icon__ i,
.c11 span.__rdt_custom_sort_icon__ svg {
  opacity: 0;
  color: inherit;
  font-size: 18px !important;
  height: 18px !important;
  width: 18px !important;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-transition-duration: 125ms;
  transition-duration: 125ms;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c11 span.__rdt_custom_sort_icon__.asc i,
.c11 span.__rdt_custom_sort_icon__.asc svg {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c11:hover {
  cursor: pointer;
}

.c11:hover span,
.c11:hover span.__rdt_custom_sort_icon__ * {
  opacity: 1;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  position: relative;
  width: 100%;
  border-radius: inherit;
  overflow-x: auto;
  overflow-y: hidden;
  min-height: 0;
}

.c3 {
  position: relative;
  width: 100%;
  display: table;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <h1
          class="c1"
        >
          delegations.activeDelegations
        </h1>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <div
              class="c4 rdt_Table"
              role="table"
            >
              <div
                class="c5 rdt_TableHead"
                role="rowgroup"
              >
                <div
                  class="c6 rdt_TableHeadRow"
                  role="row"
                >
                  <div
                    class="c7 c8 rdt_TableCol"
                  />
                  <div
                    class="c7 c8 rdt_TableCol"
                  />
                  <div
                    class="c7 c9 rdt_TableCol"
                  >
                    <div
                      class="c10 rdt_TableCol_Sortable"
                      id="column-undefined"
                      role="columnheader"
                      tabindex="0"
                    >
                      <div>
                        validator.name
                      </div>
                    </div>
                  </div>
                  <div
                    class="c7 c9 rdt_TableCol"
                  >
                    <div
                      class="c11 rdt_TableCol_Sortable"
                      id="column-amount"
                      role="columnheader"
                      tabindex="0"
                    >
                      <div>
                        delegations.delegatedAmount
                      </div>
                      <span
                        class="asc __rdt_custom_sort_icon__"
                      >
                        <svg
                          aria-label="Down"
                          class="c12"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="m2 8.35 10.173 9.823L21.997 8"
                            fill="none"
                            stroke="#000"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                  <div
                    class="c7 c13 rdt_TableCol"
                  >
                    <div
                      class="c11 rdt_TableCol_Sortable"
                      id="column-fee"
                      role="columnheader"
                      tabindex="0"
                    >
                      <div>
                        validator.fee
                      </div>
                      <span
                        class="asc __rdt_custom_sort_icon__"
                      >
                        <svg
                          aria-label="Down"
                          class="c12"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="m2 8.35 10.173 9.823L21.997 8"
                            fill="none"
                            stroke="#000"
                            stroke-width="2"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="c14 rdt_TableBody"
                offset="250px"
                role="rowgroup"
              >
                <div
                  class="c15 rdt_TableRow"
                  id="row-0"
                  role="row"
                >
                  <div
                    class="c16 c8 c17 rdt_TableCell"
                    data-tag="allowRowEvents"
                    id="cell-icon-undefined"
                    role="gridcell"
                  >
                    <img
                      alt=""
                      height="16px"
                      src="/logo192.png"
                      width="16px"
                    />
                  </div>
                  <div
                    class="c16 c8 c17 rdt_TableCell"
                    data-tag="allowRowEvents"
                    id="cell-status-undefined"
                    role="gridcell"
                  >
                    <svg
                      aria-label="StatusGood"
                      class="c18"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zM7 12l4 3 5-7"
                        fill="none"
                        stroke="#000"
                        stroke-width="2"
                      />
                    </svg>
                  </div>
                  <div
                    class="c16 c9 c17 rdt_TableCell"
                    data-tag="allowRowEvents"
                    id="cell-name-undefined"
                    role="gridcell"
                  >
                    test-validator
                  </div>
                  <div
                    class="c16 c9 c17 rdt_TableCell"
                    data-tag="allowRowEvents"
                    id="cell-amount-undefined"
                    role="gridcell"
                  >
                    0.0
                     
                    
                  </div>
                  <div
                    class="c16 c13 c17 rdt_TableCell"
                    data-tag="allowRowEvents"
                    id="cell-fee-undefined"
                    role="gridcell"
                  >
                    7.000000000000001%
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="StyledBox-sc-13pk1d4-0 huNNsk"
    >
      <h1
        class="StyledHeading-sc-1rdh4aw-0 ghJfNG"
      >
        delegations.activeDelegations
      </h1>
      <div
        class="sc-hHfuMS eHijxz"
      >
        <div
          class="sc-kfzBvY gVVUtR"
        >
          <div
            class="sc-bdfBQB dVIjvR rdt_Table"
            role="table"
          >
            <div
              class="sc-gsTEea hbMOSD rdt_TableHead"
              role="rowgroup"
            >
              <div
                class="sc-dlfnuX krZTYY rdt_TableHeadRow"
                role="row"
              >
                <div
                  class="sc-hKgJUU sc-eCstlR sc-iqHYmW eDBJAz eceSDa fTUNZi rdt_TableCol"
                />
                <div
                  class="sc-hKgJUU sc-eCstlR sc-iqHYmW eDBJAz eceSDa fTUNZi rdt_TableCol"
                />
                <div
                  class="sc-hKgJUU sc-eCstlR sc-iqHYmW eDBJAz iurrS fTUNZi rdt_TableCol"
                >
                  <div
                    class="sc-crrszt itItDX rdt_TableCol_Sortable"
                    id="column-undefined"
                    role="columnheader"
                    tabindex="0"
                  >
                    <div>
                      validator.name
                    </div>
                  </div>
                </div>
                <div
                  class="sc-hKgJUU sc-eCstlR sc-iqHYmW eDBJAz iurrS fTUNZi rdt_TableCol"
                >
                  <div
                    class="sc-crrszt eGSUoe rdt_TableCol_Sortable"
                    id="column-amount"
                    role="columnheader"
                    tabindex="0"
                  >
                    <div>
                      delegations.delegatedAmount
                    </div>
                    <span
                      class="asc __rdt_custom_sort_icon__"
                    >
                      <svg
                        aria-label="Down"
                        class="StyledIcon-sc-ofa7kd-0 jjvRzN"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m2 8.35 10.173 9.823L21.997 8"
                          fill="none"
                          stroke="#000"
                          stroke-width="2"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
                <div
                  class="sc-hKgJUU sc-eCstlR sc-iqHYmW eDBJAz hSZcZi fTUNZi rdt_TableCol"
                >
                  <div
                    class="sc-crrszt eGSUoe rdt_TableCol_Sortable"
                    id="column-fee"
                    role="columnheader"
                    tabindex="0"
                  >
                    <div>
                      validator.fee
                    </div>
                    <span
                      class="asc __rdt_custom_sort_icon__"
                    >
                      <svg
                        aria-label="Down"
                        class="StyledIcon-sc-ofa7kd-0 jjvRzN"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m2 8.35 10.173 9.823L21.997 8"
                          fill="none"
                          stroke="#000"
                          stroke-width="2"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-dIUeWJ gUlcRh rdt_TableBody"
              offset="250px"
              role="rowgroup"
            >
              <div
                class="sc-jrAFXE XoGWB rdt_TableRow"
                id="row-0"
                role="row"
              >
                <div
                  class="sc-hKgJUU sc-eCstlR sc-jSgvzq iFEPIi eceSDa gSqEps rdt_TableCell"
                  data-tag="allowRowEvents"
                  id="cell-icon-undefined"
                  role="gridcell"
                >
                  <img
                    alt=""
                    height="16px"
                    src="/logo192.png"
                    width="16px"
                  />
                </div>
                <div
                  class="sc-hKgJUU sc-eCstlR sc-jSgvzq iFEPIi eceSDa gSqEps rdt_TableCell"
                  data-tag="allowRowEvents"
                  id="cell-status-undefined"
                  role="gridcell"
                >
                  <svg
                    aria-label="StatusGood"
                    class="StyledIcon-sc-ofa7kd-0 hfdlHh"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zM7 12l4 3 5-7"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </div>
                <div
                  class="sc-hKgJUU sc-eCstlR sc-jSgvzq iFEPIi iurrS gSqEps rdt_TableCell"
                  data-tag="allowRowEvents"
                  id="cell-name-undefined"
                  role="gridcell"
                >
                  test-validator
                </div>
                <div
                  class="sc-hKgJUU sc-eCstlR sc-jSgvzq iFEPIi iurrS gSqEps rdt_TableCell"
                  data-tag="allowRowEvents"
                  id="cell-amount-undefined"
                  role="gridcell"
                >
                  0.0
                   
                  
                </div>
                <div
                  class="sc-hKgJUU sc-eCstlR sc-jSgvzq iFEPIi hSZcZi gSqEps rdt_TableCell"
                  data-tag="allowRowEvents"
                  id="cell-fee-undefined"
                  role="gridcell"
                >
                  7.000000000000001%
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
